src/rust/receiver-mock/src/metadata.rs

```diff
@@ -1,8 +1,45 @@
+use actix_web::http::header::HeaderMap;
 use anyhow::anyhow;
 use std::collections::HashMap;
 
 pub type Metadata = HashMap<String, String>;
 
+// Get the metadata from Sumo's common http headers
+// This does not include X-Sumo-Fields, which only applies to logs and is handled separately
+pub fn get_common_metadata_from_headers(headers: &HeaderMap) -> Result<Metadata, anyhow::Error> {
+    let mut metadata = Metadata::new();
+
+    // these metadata field names follow what Sumo itself does
+    // see: https://help.sumologic.com/05Search/Get-Started-with-Search/Search-Basics/Built-in-Metadata#built-in-metadata-fields
+    // and: https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/HTTP-Source/Upload-Data-to-an-HTTP-Source#supported-http-headers
+
+    match headers.get("x-sumo-name") {
+        Some(header_value) => match header_value.to_str() {
+            Ok(header_value_str) => metadata.insert("_sourceName".to_string(), header_value_str.to_string()),
+            Err(_) => return Err(anyhow!("Couldn't parse X-Sumo-Name header value")),
+        },
+        None => None,
```

Comment:
I'm trying to figure out what this means exactly. If we get to this branch does this mean... that we return None? None doesn't implement `Result` so does that just mean that we pass through to the next statement?

Reply:
Maybe that's a sign that this should really be a simple `if` statement. The `None` is the return value of the match statement. It doesn't really matter what it is, as we don't assign it to anything, but it has to be consistent with what the compiler believes is the return type - in this case it's `Option<String>`, the return type of https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.insert.